service/orchestrator/certificates.go

```diff
@@ -0,0 +1,172 @@
+package orchestrator
+
+import (
+	"context"
+	"errors"
+
+	"clouditor.io/clouditor/api/orchestrator"
+	"clouditor.io/clouditor/internal/logging"
+	"clouditor.io/clouditor/persistence"
+	"clouditor.io/clouditor/service"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+const certificationNotFoundErrorMessage = "certificate not found"
+
+// CreateCertificate implements method for creating a new certificate
+func (svc *Service) CreateCertificate(ctx context.Context, req *orchestrator.CreateCertificateRequest) (
+	res *orchestrator.Certificate, err error) {
+
+	// Validate request
+	if err = service.ValidateRequest(req); err != nil {
+		err = status.Error(codes.InvalidArgument, err.Error())
```

Comment:
The `service.ValidateRequest` method already returns a gRPC error so you must directly return the error here.